smartpy/optimizers/sgd.py

```diff
@@ -0,0 +1,51 @@
+import theano
+import theano.tensor as T
+from collections import OrderedDict
+
+import numpy as np
+from . import Optimizer
+
+
+class SGD(Optimizer):
+    def __init__(self, model, loss_fct, dataset, batch_size=1):
+        super(SGD, self).__init__(model, loss_fct, dataset)
+        self.batch_size = batch_size
```

Comment:
Make a `Batch` object that will use advanced indexing to keep track internally of the batch size.